mantidproject/mantid
# Add Settings Menu to Engineering Diffraction 2 GUI

**Description of work.**
A new settings menu has been added to the engineering diffraction 2 GUI for options that are rarely changed between uses. The settings are saved using the `mantidworkbench.ini` file created by `QSettings`. Currently, the user has options to change the location where the files generated by the GUI are saved and if they wish to force the recalculation of vanadium files, rather than attempting to load them from the chosen directory if they are available. 

Hidden from the user is an option to select a full calibration. This functionality is not yet possible as the file format generated by `EnggCalibrateFull` (`.csv` with too many columns) cannot be loaded into Mantid. Once the file can be loaded, it should be possible to simply remove the call to hide in the view. (#27659) 

**To test:**
1. Open the Engineering Diffraction GUI 
    1. As before, you may need to add `Diffraction/Engineering_Diffraction_2.py` to your properties file along with the other python GUIs.
2. Check the settings menu opens properly. 
3. Check that changing the save location works properly. 
4. Check that forcing vanadium recalculation works (it should take much longer)

Fixes #27187

*This does not require release notes* because **the GUI is not visible to users yet.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
